### PR TITLE
refactor_: add more default node config values for frontend when doing local pair sync

### PR DESCRIPTION
**Pull request by qfrank**
this is a continue work to [PR](https://github.com/status-im/status-go/pull/5080)
relate [comment](https://github.com/status-im/status-go/pull/5080#issuecomment-2076140450)

status: ready.

### CODE REVIEW

File: api/defaults.go [hunk 1]
```diff
@@ -301,23 +313,15 @@ func defaultNodeConfig(installationID string, request *requests.CreateAccount) (
...
 	}
 
-	if request.LogLevel != nil {
-		nodeConfig.LogLevel = *request.LogLevel
-		nodeConfig.LogEnabled = true
-
-	} else {
-		nodeConfig.LogEnabled = false
-	}
-
```
Comment (igor-sirotin) [inline api/defaults.go:311]: Could you please elaborate why remove this?

File: server/pairing/sync_device_test.go [hunk 2]
```diff
@@ -733,40 +729,23 @@ func defaultSettings(generatedAccountInfo generator.GeneratedAccountInfo, derive
...
 	nodeConfig.KeyStoreDir = filepath.Join(keystoreDir, keyUID)
 	nodeConfig.KeycardPairingDataFile = filepath.Join("keycard", "pairings.json")
-	nodeConfig.Name = "StatusIM"
+	nodeConfig.ShhextConfig = params.ShhextConfig{
+		InstallationID: installationID,
+	}
+
+	// need specify cluster config here, otherwise TestPairingThreeDevices will fail due to no messages(CR) received
+	// TODO(frank) need to figure out why above happen
 	clusterConfig, err := params.LoadClusterConfigFromFleet("eth.prod")
```
Comment (igor-sirotin) [inline server/pairing/sync_device_test.go]: Let's replace `eth.prod` with constant?
https://github.com/status-im/status-go/blob/34c693bdeae4de261e42d8c935c0fb8a4681f4a0/params/cluster.go#L6

Or even better take one of supported fleets:
https://github.com/status-im/status-go/blob/34c693bdeae4de261e42d8c935c0fb8a4681f4a0/params/cluster.go#L24-L29

File: server/pairing/common.go [hunk 3]
```diff
@@ -256,22 +262,94 @@ func setDefaultNodeConfig(c *params.NodeConfig) error {
...
+	specifiedNetworks := c.Networks
+	specifiedUpstreamConfigURL := c.UpstreamConfig.URL
+	specifiedLogEnabled := c.LogEnabled
+	specifiedLogLevel := c.LogLevel
+	specifiedFleet := c.ClusterConfig.Fleet
+	specifiedInstallationID := c.ShhextConfig.InstallationID
+	specifiedTorrentConfigEnabled := c.TorrentConfig.Enabled
+	specifiedTorrentConfigPort := c.TorrentConfig.Port
+
+	if len(specifiedNetworks) == 0 {
```
Comment (igor-sirotin) [inline server/pairing/common.go:282]: Maybe I'm missing something, but this looks very confusing to me.
I know you have plans on re-implementing this with smth like `login-with-syncing` request concept. I think we need to go with without merging this PR. What's the point?

We really should not pass `nodeConfig` from client side, this is wrong. And will very soon become a pain, as we want to remove `nodeConfig` handling from desktop.

-----

1. In general there's no need on passing `NodeConfig` from client.
In desktop we send a very default struct, no custom fields.

2. If we look at `ReceiverConfig`, we can see a few customizations there already, e.g. `SettingCurrentNetwork`:
https://github.com/status-im/status-go/blob/3bacb84a02406da295406acb9d8a9f9c82a25808/server/pairing/config.go#L25-L42

Let's just follow this path and add required fields to this structure. 
For sake of simplicity/readability/maintainability let's compose it into a single struct, maybe `NodeConfigOptions` (or other better name):
```go
type NodeConfigOptions struct { // find better naming
  VerifyTransactionURL string
  ...
}
```

-----

@qfrank let me know what you think
We could merge this solution and improve it in a separate PR, but I don't understand why not do it properly straightaway.

File: api/defaults.go (hunk 1 above)
Comment (qfrank) [reply api/defaults.go:311]: it's duplicated... you can see them already in line 233

File: server/pairing/sync_device_test.go (hunk 2 above)
Comment (qfrank) [reply server/pairing/sync_device_test.go]: I'll take the first since i only trust eth.prod ATM. 🙂

File: server/pairing/common.go (hunk 3 above)
Comment (qfrank) [reply server/pairing/common.go:282]: The point is: This PR has sorted out which parameters should be configurable for the frontend, and which parameters can have default values provided by the backend. The frontend no longer needs to know so many configuration details of nodeconfig(there is a mobile PR to decrease such details). Refactoring is a step-by-step process.

> We really should not pass nodeConfig from client side, this is wrong

agreed

> Let's just follow this path and add required fields to this structure.

agreed

Let's merge this and improve it in a separate PR ❤️ @igor-sirotin

**Review by Samyoul — COMMENTED** [review]
Thank you for this PR @qfrank . 🙏 

For handling NodeConfig in local syncing, we only need a small amount of parameters if any at all.

Unless the parameter is related to getting keys or db data, shouldn't config data be the responsibility of the recipient / bootstrapped device? I may be mistaken about something, but I don't think we should share the nodeconfig settings of the sending / bootstrapping device.

Please forgive me if I'm missing something.

**Review by igor-sirotin — APPROVED** [review]

**Comment by qfrank** [comment]
> Thank you for this PR @qfrank . 🙏
> 
> For handling NodeConfig in local syncing, we only need a small amount of parameters if any at all.
> 
> Unless the parameter is related to getting keys or db data, shouldn't config data be the responsibility of the recipient / bootstrapped device? I may be mistaken about something, but I don't think we should share the nodeconfig settings of the sending / bootstrapping device.
> 
> Please forgive me if I'm missing something.

yeah I agree, but currently there still are some params(e.g. OpenseaAPIKey) relate to node config need client to pass, we will remove NodeConfig and put the bare minimum needed from the client in some places as @igor-sirotin suggested before finally @Samyoul

**Review by Samyoul — APPROVED** [review]
Thank you @qfrank.

**Pull request merged**